File: implementations/node/sdk/src/contracts/publication/publication-call-site.ts
import type { AiSdkPublicationFunctionKind } from "./publication-function-kind.js";

export interface AiSdkPublicationCallSite {
  readonly kind: AiSdkPublicationFunctionKind;
  readonly stepName?: string;
  readonly policyIndex?: number;
  readonly definitionPath?: string;
}

File: implementations/python/sdk/src/multiplexed_ai_sdk/contracts/publication/publication_call_site.py
from __future__ import annotations

from dataclasses import dataclass

from ...wire_model import AiSdkWireModel
from .publication_function_kind import AiSdkPublicationFunctionKind


@dataclass(frozen=True)
class AiSdkPublicationCallSite(AiSdkWireModel):
    kind: AiSdkPublicationFunctionKind
    step_name: str | None = None
    policy_index: int | None = None
    definition_path: str | None = None
